### PR TITLE
[feature fix] Renamed pointers in v2 API to node_links

### PR DESCRIPTION
**Pull request by reinaH**
# Purpose: 
-https://github.com/CenterForOpenScience/osf.io/issues/3822 Because on the OSF pointers are called "links" to avoid confusion with JSON API links, on the API pointers should be renamed to "node_links"

# Changes: 
- Name changes in nodes/serializers.py; route changes in nodes/urls.py; url changes in nodes/test_views.py

# Side Effects:
 Affects routes, docs, and how we return data.

# Notes: 
Closes issue:
https://github.com/CenterForOpenScience/osf.io/issues/3822

### CODE REVIEW

**Comment by reinaH** [comment]
@brianjgeiger

**Comment by sloria** [comment]
Don't serializer names appear in the documentation (as the "model" for responses)? If so, then we'll want to rename the `NodePointerSerializer`.

**Comment by sloria** [comment]
This help text needs to be updated as well: https://github.com/reinaH/osf.io/blob/513d81c0bc8950011cf020b02057e569b8a32de7/api/nodes/serializers.py#L136-138

**Comment by sloria** [comment]
Similarly, do view class names and their docstrings appear in the API docs? If so, then these views need to be changed: https://github.com/reinaH/osf.io/blob/513d81c0bc8950011cf020b02057e569b8a32de7/api/nodes/views.py#L197-201

**Comment by sloria** [comment]
Passing this back to @reinaH . Please do a thorough search of the API for anywhere where "pointer" might be exposed to a user and rename it to use the new language.

Once that is done, ping @brianjgeiger to for review.

**Comment by brianjgeiger** [comment]
You don't have to close it, just make the changes and push to update the existing PR.

File: api/nodes/serializers.py [hunk 1]
```diff
@@ -130,15 +130,15 @@ def update(self, instance, validated_data):
         return instance
 
 
-class NodePointersSerializer(JSONAPISerializer):
+class NodeLinksSerializer(JSONAPISerializer):
```
Comment (brianjgeiger) [inline api/nodes/serializers.py]: Strictly speaking, these should probably be NodeNodeLinks, but that sounds dumb. Let's try it this way for the moment.

Comment (reinaH) [inline api/nodes/serializers.py]: i had noticed that also. but it looked odd...

File: api/nodes/views.py [hunk 2]
```diff
@@ -194,38 +194,38 @@ def get_queryset(self):
         return children
 
 
-class NodePointersList(generics.ListCreateAPIView, NodeMixin):
-    """Pointers to other nodes.
+class NodeLinksList(generics.ListCreateAPIView, NodeMixin):
+    """Node_links to other nodes.
```
Comment (brianjgeiger) [inline api/nodes/views.py]: I think in the docstrings we should just call them `Node Links` unless it's referring specifically to a parameter or path component.

File: api/nodes/views.py [hunk 3]
```diff
@@ -194,38 +194,38 @@ def get_queryset(self):
...
 
-    Pointers are essentially aliases or symlinks: All they do is point to another node.
+    Node_links are essentially aliases or symlinks: All they do is point to another node.
```
Comment (brianjgeiger) [inline api/nodes/views.py]: Node Links

File: api/nodes/views.py [hunk 4]
```diff
@@ -194,38 +194,38 @@ def get_queryset(self):
...
 
     def get_queryset(self):
         pointers = self.get_node().nodes_pointer
         return pointers
 
 
-class NodePointerDetail(generics.RetrieveDestroyAPIView, NodeMixin):
-    """Detail of a pointer to another node.
+class NodeLinksDetail(generics.RetrieveDestroyAPIView, NodeMixin):
+    """Detail of a node_link to another node.
```
Comment (brianjgeiger) [inline api/nodes/views.py]: Node Link

File: api/nodes/views.py [hunk 5]
```diff
@@ -194,38 +194,38 @@ def get_queryset(self):
...
 
-    Pointers are essentially aliases or symlinks: All they do is point to another node.
+    Node_links are essentially aliases or symlinks: All they do is point to another node.
```
Comment (brianjgeiger) [inline api/nodes/views.py]: Node Links

**Comment by brianjgeiger** [comment]
Pass done. :penguin:  After these small docstring changes it can go to Steve.

**Comment by reinaH** [comment]
@sloria

**Pull request merged**